doc/neps/nep-0037-array-module.rst

```diff
@@ -26,32 +26,28 @@ Why ``__array_function__`` hasn't been enough
 
 There are two broad ways in which NEP-18 has fallen short of its goals:
 
-1. **Maintainability concerns**. `__array_function__` has significant
+1. **Backwards compatibility concerns**. `__array_function__` has significant
    implications for libraries that use it:
 
-   - Projects like `PyTorch
-     <https://github.com/pytorch/pytorch/issues/22402>`_, `JAX
-     <https://github.com/google/jax/issues/1565>`_ and even `scipy.sparse
-     <https://github.com/scipy/scipy/issues/10362>`_ have been reluctant to
-     implement `__array_function__` in part because they are concerned about
-     **breaking existing code**: users expect NumPy functions like
+   - `JAX <https://github.com/google/jax/issues/1565>`_ has been reluctant
+     to implement ``__array_function__`` in part because it is concerned about
+     breaking existing code: users expect NumPy functions like
      ``np.concatenate`` to return NumPy arrays. This is a fundamental
      limitation of the ``__array_function__`` design, which we chose to allow
      overriding the existing ``numpy`` namespace.
+
+     Note that projects like `PyTorch
+     <https://github.com/pytorch/pytorch/issues/22402>`_ and `scipy.sparse
+     <https://github.com/scipy/scipy/issues/10362>`_ have also not
+     adopted ``__array_function__`` yet, because they don't have a
+     NumPy-compatible API or semantics. In the case of PyTorch, that is likely
+     to be added in the future; ``scipy.sparse`` is in the same situation as
```

Comment:
Would it be a fair summary to say something like "PyTorch prefers not to break backwards compatibility but is likely to add `__array_function__` in the future anyways"?

I'm trying to get across that these backwards compatibility concerns are not unique to JAX, even if the overall assessment of cost/benefits varies.

Reply:
> Would it be a fair summary to say something like "PyTorch prefers not to break backwards compatibility but is likely to add `__array_function__` in the future anyways"?

I think that statement is true but also misleading. There has been no concern AFAIK about backwards compat in this regard for PyTorch. The issue is purely that there is no NumPy-compatible API.

> I'm trying to get across that these backwards compatibility concerns are not unique to JAX, even if the overall assessment of cost/benefits varies.

Then let's say that instead. Something like "Libraries like Dask and CuPy have looked at and accepted the backwards incompatibility impact of `__array_function__`; it would still have been better for them if that impact didn't exist".

Reply:
> I think that statement is true but also misleading. There has been no concern AFAIK about backwards compat in this regard for PyTorch. The issue is purely that there is no NumPy-compatible API.

I am happy to let this drop, but just so you understand where these words came from, here are things I saw you write last summer about `__array_function__` in PyTorch:

- "There may be a backwards compatibility issue here" https://github.com/pytorch/pytorch/issues/22402
- "Implementing it has some backwards compat concerns as well, because people may be relying on `np.somefunc(some_torch_tensor)` to be coerced to `ndarray`."  https://github.com/pydata/xarray/issues/3232#issuecomment-522824210

> Then let's say that instead. Something like "Libraries like Dask and CuPy have looked at and accepted the backwards incompatibility impact of `__array_function__`; it would still have been better for them if that impact didn't exist".

👍

Reply:
> I am happy to let this drop, but just so you understand where these words came from, here are things I saw you write last summer about `__array_function__` in PyTorch:

Yeah I know, since it is something that needs considering. The proposal was still received positively though; as soon as there's a NumPy compatible API (which will take quite a while, there's complicated semantic incompatibilities like `axis=None`) I expect `__array_function__` to be adopted. Which is covered by my proposed rephrase.

Reply:
> > I am happy to let this drop, but just so you understand where these words came from, here are things I saw you write last summer about `__array_function__` in PyTorch:
> 
> Yeah I know, I wrote it since it is something that needs considering. The proposal was still received positively though; as soon as there's a NumPy compatible API (which will take quite a while, there's complicated semantic incompatibilities like `axis=None`) I expect `__array_function__` to be adopted. Which is covered by my proposed rephrase.

Reply:
adopted the suggested phrasing